### ISSUE TITLE
questions about learned disparity.

### ISSUE THREAD

**Issue by philipjd** [mrharicot/monodepth#73]
Hi, I recently studied your great work and read your code. But there are something I can't understand well.
In your paper and your repo, you mentioned that you have models pretrained on Cityscapes and fine-tuned on Kitti. But these two datasets are based on different stereo pairs, i.e. the focal length & base are different for these two datasets, which means the same scenes for these two stereo pairs would have different disparities in images. 
So my question is, would the fine-turning on one dataset while pretrained on the other confuse the model training? Since the disparities would have different scales for these datasets.
And one more question, I saw in your code, in "get_disp" function, you calculate the disparity by multiplying a 0.3 to the output of conv layer. How does this magic number 0.3 comes out?

Thanks a lot for your time.

**Comment by mrharicot** [comment]
Hi,
That's an excellent question!
Fine tuning on a different dataset did not seem to be an issue as we train on only a dataset at once, the network learns to change the scale very quickly it seems.
If we were training on two datasets at once then the network would either get confused, or would have to detect what dataset the image is from.

Finally the magic number 0.3 came from me eyeballing the maximum disparity in the KITTI images, which I found to be ~15% of the width, so I used twice this value to be the maximum the disparity could reach.
As it turns out the maximum disparity is even lower and closer to 8% iirc.

I hope this helps!
